src/porepy/models/boundary_condition.py

```diff
@@ -92,24 +95,22 @@ def update_boundary_condition(
                 # No previous time step exists. The method was called during
                 # the initialization.
                 vals = function(bg)
-            pp.set_solution_values(name=name, values=vals, data=data, time_step_index=1)
+
+            # Before setting the new, most recent time step, shift the stored values
+            # backwards in time
+            pp.shift_solution_values(
```

Comment:
I enhanced the functionality of the BoundaryConditions mixin, to now support the full time step depth.
Removed the warning below.